File: 链表/JZ24.py
# 反转链表

# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
#
# 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
#
# 
# @param head ListNode类 
# @return ListNode类

class Solution:
    def ReverseList(self , head: ListNode) -> ListNode:
        # write code here
        pnode = ListNode(None)
        pnode = head
        if head == None or head.next == None:
            return pnode
        qnode = pnode.next
        while qnode != None:
            head.next = qnode.next
            qnode.next = pnode
            pnode = qnode
            qnode = head.next
        head.next = None 
        return pnode

File: 链表/JZ24.cpp
#include <cstddef>
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
		ListNode* pnode = pHead;
		if(pHead==NULL || pHead->next==NULL) {
			return pnode;
		}
		ListNode* qnode = pnode->next;
		while(qnode!=NULL){
			pHead->next=qnode->next;
			qnode->next = pnode;
			pnode = qnode;
			qnode = pHead->next;
		}
		return pnode;
    }
};
